src/Sarif.Converters/FortifyFprConverter.cs

```diff
@@ -477,6 +487,41 @@ private void ParseVulnerability(IResultLogWriter output)
             output.WriteResult(result);
         }
 
+        internal static FailureLevel GetFailureLevelFromRuleMetadata(ReportingDescriptor rule)
+        {
+            // High impact - ["5.0", "3.0") => Error
+            // Medium impact - ["3.0", "1.0") => Warning
+            // Low impact - ["1.0", "0,0"] => Note
+            // Any other value (invalid input) or no value => None
+
+            if (rule.TryGetProperty("Impact", out string impactValue))
+            {
+                if (float.TryParse(impactValue, out float impact))
+                {
+                    if (impact > 5.0)
+                    {
+                        return FailureLevel.None;
```

Comment:
>FailureLevel.None; [](start = 31, length = 18)

to use a very uncouth american phrase, i pulled it out my... no i cant say it i am sorry.

I just randomly decided that any invalid value (ie. not [0 - 5.0] should be considered "None". I think the alternate is that this value should be left unset -> which means it would get the default value of "Warning". So, my intent here is to say - NO, we didnt know what the severity of this issue is because it wasnt provided!


..just tell me. Thank you.

Reply:
I don't agree with this. The result would be that if somebody made a mistake in the definition of a Fortify rule, that end users would tend to ignore all results from that rule, even if the rule is pointing out a serious problem! (Automated systems might filter out results with level "none", and then end users might not see them at all!)

- I think you should treat impact > 5.0 as "error" (maybe the rule author forgot that 5.0 was the max, and they turned it up to "11" :-))

- I think you should treat impact < 0.0 as "warning". At least get the result on the end-user's radar.

- If impact is missing, I think you should also treat that as "warning". See my comment in the unit test code for my reasoning.

---
In reply to: [292262904](https://github.com/microsoft/sarif-sdk/pull/1528#discussion_r292262904) [](ancestors = 292262904)

Reply:
Thanks Larry! 

---
In reply to: [292551572](https://github.com/microsoft/sarif-sdk/pull/1528#discussion_r292551572) [](ancestors = 292551572,292262904)